numpy/lib/function_base.py

```diff
@@ -1522,17 +1531,81 @@ def gradient(f, *varargs, **kwargs):
     -------
     gradient : ndarray or list of ndarray
         A set of ndarrays (or a single ndarray if there is only one dimension)
-        correposnding to the derivatives of f with respect to each dimension.
+        corresponding to the derivatives of f with respect to each dimension.
         Each derivative has the same shape as f.
 
+    Notes
+    -----
+    Assuming that :math:`f\in C^{3}` and let be :math:`h_{*}` a non homogeneous 
```

Comment:
Can you use words for `C^{3}` instead of symbols? This is hard to Google for.

Maybe something like:

    Assume f is a triply continuous differentiable function and let `h_{*}` be a
    non-homogeneous stepsize.
    The finite-difference formula used here can be derived by
    minimizing the consistency error `\eta{i}`: